### PR TITLE
Add CentOS 8 Azure Configuration

### PR DESCRIPTION
**Pull request by szarkos**

### CODE REVIEW

File: cloudimg/CentOS-8-x86_64-Azure.ks [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+-biosdevname
+-plymouth
+-iprutils
+
+# enable rootfs resize on boot
+cloud-utils-growpart
+gdisk
+
+# add insight-clients
+insights-client
```
Comment (kbsingh) [inline cloudimg/CentOS-8-x86_64-Azure.ks]: is this going to work on CentOS ?

File: cloudimg/CentOS-8-x86_64-Azure.ks [hunk 2]
```diff
@@ -0,0 +1,200 @@
+# Kickstart for creating a CentOS 8 Azure VM
+
+# System authorization information
+auth --enableshadow --passalgo=sha512
+
+# Use graphical install
+text
```
Comment (johanburati) [inline cloudimg/CentOS-8-x86_64-Azure.ks]: Comment should be *# Use text install*

File: cloudimg/CentOS-8-x86_64-Azure.ks [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+dnf clean all
+
+# XXX instance type markers - MUST match CentOS Infra expectation
+echo 'azure' > /etc/yum/vars/infra
+
+# Set tuned profile
+echo "virtual-guest" > /etc/tuned/active_profile
+
+# Deprovision and prepare for Azure
+/usr/sbin/waagent -force -deprovision
```
Comment (johanburati) [inline cloudimg/CentOS-8-x86_64-Azure.ks]: The image has not been provisioned, is there any reason of doing this ?

Comment (szarkos) [reply cloudimg/CentOS-8-x86_64-Azure.ks]: Yes - this is in accordance with the official docs from the Azure agent. It probably doesn't do much, but maybe ensure that the root account is locked, and maybe other things that I'm not aware of.

File: cloudimg/CentOS-8-x86_64-Azure.ks (hunk 2 above)
Comment (szarkos) [reply cloudimg/CentOS-8-x86_64-Azure.ks]: Will do - thanks!

File: cloudimg/CentOS-8-x86_64-Azure.ks (hunk 1 above)
Comment (szarkos) [reply cloudimg/CentOS-8-x86_64-Azure.ks]: No - I removed

**Comment by szarkos** [comment]
I'd like to make bigger changes to this kickstart to support generation-2 VMs in Azure. This will ensure support for things like Azure Confidential Computing, Mv2 sizes, some HPC VM sizes, and eventually things like SecureBoot.

I'm going to close this PR and open a new one to review, and I'll integrate feedback from here as well.

**Pull request closed without merging**